tutorials/authoring-recipes/writing-a-java-refactoring-recipe.md

```diff
@@ -19,46 +20,51 @@ And refactor it into a class like this:
+## Prerequisites
+
+This guide assumes you've already set up your [Recipe Development Environment](../../getting-started/recipe-development-environment.md) and that you are familiar with writing Java code.
+
+## Outlining SayHelloRecipe
 
-## Defining SayHelloRecipe
+Let's begin by creating a Java class that extends `org.openrewrite.Recipe`. At a minimum, this recipe should have:
 
-Begin by creating a class that extends `org.openrewrite.Recipe`. This recipe should accept as a configuration parameter the fully qualified name of the class to add a `hello()` method to and it should validate that parameter is configured with a valid value.
+* An `Option` that is the fully qualified name of the class you want to add the `hello()` method to. 
+* A serializable constructor that takes in the fully qualified class name.
+* A `getDisplayName()` method that returns the display name for this recipe.
+* A `getDescription()` method that returns the description for this recipe.
+
+Here is what the class should look like with just these things defined:
 
 ```java
-package org.openrewrite.samples;
+package com.yourorg;
 
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
-import org.openrewrite.ExecutionContext;
+import lombok.EqualsAndHashCode;
+import lombok.Value;
 import org.openrewrite.Option;
 import org.openrewrite.Recipe;
 import org.openrewrite.internal.lang.NonNull;
-import org.openrewrite.java.JavaIsoVisitor;
-import org.openrewrite.java.JavaTemplate;
-import org.openrewrite.java.tree.J;
 
+// Making your recipe immutable helps make them idempotent and eliminates a variety of possible bugs.
+// Configuring your recipe in this way also guarantees that basic validation of parameters will be done for you by rewrite.
+@Value
+@EqualsAndHashCode(callSuper = false)
 public class SayHelloRecipe extends Recipe {
-    // Making your recipe immutable helps make them idempotent and eliminates categories of possible bugs
-    // Configuring your recipe in this way also guarantees that basic validation of parameters will be done for you by rewrite
     @Option(displayName = "Fully Qualified Class Name",
-            description = "A fully-qualified class name indicating which class to add a hello() method.",
+            description = "A fully qualified class name indicating which class to add a hello() method to.",
             example = "com.yourorg.FooBar")
     @NonNull
-    private final String fullyQualifiedClassName;
-    public String getFullyQualifiedClassName() {
-        return fullyQualifiedClassName;
-    }
+    String fullyQualifiedClassName;
 
-    // Recipes must be serializable. This is verified by RecipeTest.assertChanged() and RecipeTest.assertUnchanged()
+    // All recipes must be serializable. This is verified by RewriteTest.rewriteRun() in your tests.
```

Comment:
Pretty sure you can remove this manually created constructor, it will be generated via Lombok's `@Value` annotation. You do not need the `JsonProperty` annotation if you add `-parameters` as a compiler option.

Reply:
I think it would be enough to say that "A recipe's state is serialized to/from JSON using the Jackson serialization library. The serializability of a recipe is verified by RewriteTest.rewriteRun() in your tests."

Reply:
@tkvangorder Thanks for the call out! I'm not sure what you mean by `-parameters` as a compiler option, though. Would you mind expanding on what people would need to do for that? I saw that my tests ran just fine if I removed the constructor as you suggested so I'm happy to update the docs and remove it altogether. Unsure if there's a concern with that, though. Maybe I'll add the serialize comment to the comment above the lombok annotation?